lib/src/utils/crypto-utils.ts

```diff
@@ -1,182 +0,0 @@
- *
- * WSO2 Inc. licenses this file to you under the Apache License,
- * Version 2.0 (the "License"); you may not use this file except
- * in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied. See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-import base64url from "base64url";
-import WordArray from "crypto-js/lib-typedarrays";
-import sha256 from "fast-sha256";
-// Importing from node_modules since rollup doesn't support export attribute of `package.json` yet.
-import randombytes from "randombytes";
-import parseJwk from "../../node_modules/jose/dist/browser/jwk/parse";
-import jwtVerify, { KeyLike } from "../../node_modules/jose/dist/browser/jwt/verify";
-import { RuntimeEnvironments } from "../constants";
-import { AsgardeoAuthException } from "../exception";
-import { DecodedIDTokenPayload, JWKInterface } from "../models";
-const nodeRandomBytes = require("secure-random-bytes");
-
-export class CryptoUtils {
-    // eslint-disable-next-line @typescript-eslint/no-empty-function
-    private constructor() {}
-    /**
-     * Get URL encoded string.
-     *
-     * @param {CryptoJS.WordArray} value.
-     * @returns {string} base 64 url encoded value.
-     */
-    public static base64URLEncode(value: Buffer | string): string {
-        return base64url.encode(value).replace(/\+/g, "-").replace(/\//g, "_").replace(/=/g, "");
-    }
-
-    /**
-     * Generate code verifier.
-     *
-     * @returns {string} code verifier.
-     */
-    public static getCodeVerifier(env :string): string {
-        switch(env){
-            case RuntimeEnvironments.Node:
-                return this.base64URLEncode(nodeRandomBytes(32));
-            case RuntimeEnvironments.Browser:
-                return this.base64URLEncode(randombytes(32));
-            case RuntimeEnvironments.ReactNative:
-                return this.base64URLEncode(WordArray.random(32).toString());
-            default:
-                return this.base64URLEncode(randombytes(32)); //Default fallback is the browser.
-        }
-    }
-
-    /**
-     * Derive code challenge from the code verifier.
-     *
-     * @param {string} verifier.
-     * @returns {string} code challenge.
-     */
-    public static getCodeChallenge(verifier: string): string {
-        return this.base64URLEncode(new Buffer(sha256(new TextEncoder().encode(verifier))));
-    }
-
-    /**
-     * Get the supported signing algorithms for the id_token.
-     *
-     * @returns {string[]} array of supported algorithms.
-     */
-    public static getSupportedSignatureAlgorithms(): string[] {
-        return ["RS256", "RS512", "RS384", "PS256"];
-    }
```

Comment:
This can be provided by this util itself.

Reply:
Moved to `src/constants` as discussed offline